lib/linux_admin/ssh_agent.rb

```diff
@@ -0,0 +1,64 @@
+require 'net/ssh'
+require 'fileutils'
+module LinuxAdmin
+  class SSHAgent
+    attr_accessor :pid
+    attr_reader :socket
+
+    def initialize(ssh_private_key, options = {})
+      @socket = options[:agent_socket] if options[:agent_socket]
+      @private_key = ssh_private_key
+    end
+
+    def start
+      if @socket
+        FileUtils.mkdir_p(File.dirname(@socket))
+        agent_details = `ssh-agent -a #{@socket}`
+      else
+        agent_details = `ssh-agent`
+        @socket = parse_ssh_agent_socket agent_details
+      end
+      @pid = parse_ssh_agent_pid agent_details
+      IO.popen({ 'SSH_AUTH_SOCK' => @socket, 'SSH_AGENT_PID' => @pid }, ['ssh-add', '-'], mode: 'w') do |f|
+        f.puts(@private_key)
+      end
+      raise StandardError, "Couldn't add key to agent" if $CHILD_STATUS.to_i != 0
+    end
+
+    def with_service(_options = {})
+      start
+      yield @socket
+      stop
+    rescue => e
+      stop
+      raise e
+    end
+
+    def stop
+      system({ 'SSH_AGENT_PID' => @pid }, '(ssh-agent -k) &> /dev/null') if process_exists?(@pid)
```

Comment:
This could also be `Common.run!(Common.cmd("ssh-agent"), :env => {"SSH_AGENT_PID" => @pid}, :params => [:k]) if process_exists?(@pid)`

That would allow the logger to be used, dynamically find the absolute path to the `ssh-agent` binary, and still capture the output.